rust/signed_doc/src/metadata/extra_fields.rs

```diff
@@ -196,41 +154,57 @@ impl ExtraFields {
             COSE_DECODING_CONTEXT,
             error_report,
         );
-        let brand_id = decode_document_field_from_protected_header(
-            protected,
-            BRAND_ID_KEY,
-            COSE_DECODING_CONTEXT,
-            error_report,
-        );
-        let campaign_id = decode_document_field_from_protected_header(
-            protected,
-            CAMPAIGN_ID_KEY,
-            COSE_DECODING_CONTEXT,
-            error_report,
-        );
-        let election_id = decode_document_field_from_protected_header::<CborUuidV4>(
-            protected,
-            ELECTION_ID_KEY,
-            COSE_DECODING_CONTEXT,
-            error_report,
-        )
-        .map(|v| v.0);
-        let category_id = decode_document_field_from_protected_header(
-            protected,
-            CATEGORY_ID_KEY,
-            COSE_DECODING_CONTEXT,
-            error_report,
-        );
+        let parameters = {
+            // process `parameters` field and all its aliases
+            let parameters: Option<DocumentRef> = decode_document_field_from_protected_header(
```

Comment:
I feel like these calls, could be simplified to iterate on a list of `[PARAMETERS_KEY, BRAND_ID_KEY, CAMPAIGN_ID_KEY, CATEGORY_ID_KEY]` and directly produce the `[parameters, brand_id, campaign_id, category_id]` array.
Which might be less repetitive, and probably eliminate the need for `too_many_lines` lint being allowed above?

Reply:
yea, you are right, updated